modules/proxy/mod_proxy_hcheck.c

```diff
@@ -509,12 +528,17 @@ static proxy_worker *hc_get_hcworker(sctx_t *ctx, proxy_worker *worker,
         hc->hash.def = hc->s->hash.def = ap_proxy_hashfunc(hc->s->name, PROXY_HASHFUNC_DEFAULT);
         hc->hash.fnv = hc->s->hash.fnv = ap_proxy_hashfunc(hc->s->name, PROXY_HASHFUNC_FNV);
         hc->s->port = port;
-        hc->s->conn_timeout_set = worker->s->conn_timeout_set;
-        hc->s->conn_timeout = worker->s->conn_timeout;
-        hc->s->ping_timeout_set = worker->s->ping_timeout_set;
-        hc->s->ping_timeout = worker->s->ping_timeout;
-        hc->s->timeout_set = worker->s->timeout_set;
-        hc->s->timeout = worker->s->timeout;
+        if(worker->s->hc_timeout_set) {
+            hc->s->timeout = worker->s->hc_timeout;
+            hc->s->timeout_set = 1;
+        } else {
+            hc->s->conn_timeout_set = worker->s->conn_timeout_set;
+            hc->s->conn_timeout = worker->s->conn_timeout;
+            hc->s->ping_timeout_set = worker->s->ping_timeout_set;
+            hc->s->ping_timeout = worker->s->ping_timeout;
+            hc->s->timeout_set = worker->s->timeout_set;
+            hc->s->timeout = worker->s->timeout;
+        }
```

Comment:
If we don't have a specific `hc` for the other timeouts (`conn_` and `ping_`), shouldn't we keep the worker's ones at least rather than the defaults?